File: 1953/1953.cs
public class Solution {
    public int MaximumNumberOfWeeksForWhichYo(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MaximumNumberOfWeeksForWhichYo(root.left), MaximumNumberOfWeeksForWhichYo(root.right));
    }
}

File: 1953/1953.go
func maximumNumberOfWeeksForWhichYo(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := maximumNumberOfWeeksForWhichYo(root.Left)
	r := maximumNumberOfWeeksForWhichYo(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}